### DISCUSSION TITLE
Compatibility with VirtualBox Guest Additions

### DISCUSSION THREAD

**Discussion by grahamperrin** [helloSystem/hello#219] [General]
[emulators/virtualbox-ose-additions](https://www.freshports.org/emulators/virtualbox-ose-additions/)

With VBoxSVGA in use, after starting the required services: 

![image](https://user-images.githubusercontent.com/192271/137609124-2c3761aa-ac23-43e4-8bc6-1ea57da57291.png)

## Bugs

1. screen resizing is non-effective
2. the bidirectional shared clipboard does not work
3. maybe more …

Failure to share the clipboard is a significant hindrance to testing and feedback. 

Please, can anyone tell what causes these issues? 

TIA

----

Background: https://github.com/helloSystem/ISO/issues/87 (closed by me as out of scope).

**Comment by probonopd** [comment]
What happens if you install to disk, install the VirtualBox additions, and restart the whole system?

**Reply by grahamperrin** [reply]
Already installed to disk; please note the up-to-date base system with 12.2-RELEASE-p10 userland. 

A restart of the OS was probably unnecessary, but I did so before postsing. 

Origin: https://github.com/helloSystem/ISO/releases/tag/r0.6.0 preset to latest for packages.

**Reply by probonopd** [reply]
> Origin: https://github.com/helloSystem/ISO/releases/tag/r0.6.0 preset to latest for packages.

Now I am confused. I don't remember that we used `latest` packages for anything but very experimental FreeBSD 13 based builds.

**Reply by grahamperrin** [reply]
https://github.com/helloSystem/hello/discussions/219#discussioncomment-1489747

> Does 0.7.0 perform better in this regard?

No. Copied text (example below) is not shared with the host. 

![image](https://user-images.githubusercontent.com/192271/137641095-a07cb2da-7dd8-4476-a784-5d60eb2e18dd.png)

**Comment by grahamperrin** [comment]
[helloSystem 0.6.0 (12.2-RELEASE r366954): latest packages · Discussion #222 · helloSystem/hello](https://github.com/helloSystem/hello/discussions/222)

**Reply by probonopd** [reply]
True for 0.6.0 as per "Known Issues" in the Release Notes. No longer true for 0.7.0 experimental builds.
Sorry for the confusion.

**Reply by probonopd** [reply]
Does 0.7.0 perform better in this regard?

**Reply by grahamperrin** [reply]
At a glance: little or no better. 

The absence of a shared clipboard is a massive hindrance :-(

**Comment by probonopd** [comment]
I am giving VirtualBox a try on my devel system:

```
FreeBSD% freebsd-version -k -u
12.2-RELEASE-p7
12.2-RELEASE-p10
FreeBSD% pkg search virtualbox
phpvirtualbox-6.1              AJAX Web Interface for VirtualBox
phpvirtualbox-legacy-5.2.1_1   AJAX Web Interface for VirtualBox
virtualbox-ose-6.1.26_3        General-purpose full virtualizer for x86 hardware
virtualbox-ose-additions-6.1.26_1 VirtualBox additions for FreeBSD guests
virtualbox-ose-additions-legacy-5.2.44_3 VirtualBox additions for FreeBSD guests
virtualbox-ose-additions-nox11-6.1.26_1 VirtualBox additions for FreeBSD guests
virtualbox-ose-additions-nox11-legacy-5.2.44_3 VirtualBox additions for FreeBSD guests
virtualbox-ose-kmod-6.1.26     VirtualBox kernel module for FreeBSD
virtualbox-ose-kmod-legacy-5.2.44_5 VirtualBox kernel module for FreeBSD
virtualbox-ose-legacy-5.2.44_9 General-purpose full virtualizer for x86 hardware
virtualbox-ose-nox11-6.1.26_3  General-purpose full virtualizer for x86 hardware
virtualbox-ose-nox11-legacy-5.2.44_9 General-purpose full virtualizer for x86 hardware
```

So I am assuming `sudo pkg install -y virtualbox-ose virtualbox-ose-additions virtualbox-ose-kmod` is the way to go and I don't need `-legacy`? Then:

```
FreeBSD% sudo kldload /boot/modules/vboxdrv.ko 
FreeBSD% sudo -A -E VirtualBox
```

Results in:

![image](https://user-images.githubusercontent.com/2480569/138597652-0db42384-fdac-4a2c-b37d-f8f3b69d0caf.png)

When I resize the window, the virtual screen size is not resized. But the **vboxguest** kernel extension is loaded and the **VBoxService** process is running.

![image](https://user-images.githubusercontent.com/2480569/138597817-845543b2-b481-481b-a1ad-2f9eb38f75af.png)

**Reply by grahamperrin** [reply]
> … assuming `sudo pkg install -y virtualbox-ose virtualbox-ose-additions virtualbox-ose-kmod` is the way to go …

Not all three. **NB** <https://www.freshports.org/emulators/virtualbox-ose-additions/#description> "… Additions are not intended for use on Host machines. …" etc..

In the host: 

1. install `emulators/virtualbox-ose` **alone** (allow automated installation of dependencies)
2. observe the package message <https://www.freshports.org/emulators/virtualbox-ose/#message>. 

In the guest: 

1. install `emulators/virtualbox-ose-additions` **alone**
2. observe the package message <https://www.freshports.org/emulators/virtualbox-ose-additions/#message>.

**Reply by probonopd** [reply]
OK, thanks, makes sense.

**Comment by grahamperrin** [comment]
Cross-reference: https://redd.it/xd0zzc

**Comment by probonopd** [comment]
Since my hardware is not nearly powerful enough to work with virtual machines, I will not be working on this myself but if someone can send a pull request that improves the situation, I'll be happy to merge this.

**Comment by probonopd** [comment]
I know it took a while, but: VirtualBox Guest Additions are now preinstalled and enabled, providing for seamless resizing of the screen, copy & paste, etc. when running in VirtualBox. Please try the latest experimental build. Thanks for your patience! :+1:
